example-wasm/src/pages/LoginPage.tsx

```diff
@@ -0,0 +1,94 @@
+import { FormEvent, useState } from 'react';
+import { Navigate, useNavigate } from 'react-router-dom';
+
+import { useAppManager } from '../hooks/useAppManager';
+import logo from '../assets/logo.png';
+import styles from '../styles/LoginPage.module.css';
+
+const { useApp } = await import('@realm/react');
+
+export function LoginPage() {
+  const atlasApp = useApp();
+  const navigate = useNavigate();
+  const { register, logIn } = useAppManager();
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [authRequest, setAuthRequest] = useState<'login' | 'register'>('login');
+
+  if (atlasApp.currentUser) {
+    return <Navigate to='/tasks' />
+  }
+
+  const resetForm = () => {
+    setEmail('');
+    setPassword('');
+  };
+
+  const handleSubmit = async (event: FormEvent<HTMLFormElement>): Promise<void> => {
+    event.preventDefault();
+
+    try {
+      if (authRequest === 'register') {
+        await register({ email, password });
+      }
+      await logIn({ email, password });
+    } catch (err: any) {
+      return console.error(`Error ${authRequest === 'login' ? 'logging in' : 'registering'}: ${err.message || err}`);
+    }
+
+    resetForm();
+    navigate('/tasks');
```

Comment:
True, maybe it could be valuable though to show the routing? Mainly since I'd assume most React apps would set up routing to be able to provide links to various parts.